app/src/main/java/com/quizarena/menu/MainMenuActivity.kt

```diff
@@ -46,20 +47,32 @@ class MainMenuActivity : AppCompatActivity() {
             // the activity was opened by a sharing link
             // go to the needed activity
             val session = SessionSharing().getSessionFromLink(this, intent.data)
-            val detailViewIntent = Intent(this@MainMenuActivity, SessionDetailActivity::class.java)
-            detailViewIntent.putExtra(getString(R.string.intent_extra_session_clicked), session)
-            startActivity(detailViewIntent)
-
-            // override the intent action for going back in the other activity
-            this.intent.action = Intent.ACTION_MAIN
+            startSessionDetailActivity(session)
 
         } else {
-            // activity was started manually, not by a link
+            val sessionID = intent.getStringExtra("session")
+            if(sessionID != null){
+                // the activity was opened by a push notification
+                // go to the needed activity
+                val session = SessionSharing().getSessionFromID(this, sessionID)
+                startSessionDetailActivity(session)
```

Comment:
Du solltest hier besser die SessionParticipantDetailActivity (oder wie die heißt) aufrufen. 
Ich gehe beim Sharing davon aus, dass der User noch nicht an der Session teilgenommen hat.
Hier gehen wir aber davon aus, dass er die Nachricht empfängt, wenn er teilgenommen hat. Das ist dann eine andere Activity.

Noch schöner wäre es natürlich, wenn du in der startSessionDetailView-Methode prüfst ob der Nutzer Teilnehmer ist (Ist im Session Objekt bereits als boolean hinterlegt) und dann die richtige Activity aufrufst.
Dann wäre es unabhängig von Sharing und Push und wir würden uns immer die Weiterleitung sparen :)

Reply:
Ich starte jetzt die jeweils passende Activity. Die Weiterleitungen habe ich noch nicht rausgenommen. Leider bekomme ich jetzt bei der letzten Frage eine IndexOutOfBoundsException die ich mir nicht erklären kann. Hat von euch jemand eine Idee?

Reply:
Wir haben einen IndexOutOfBounds-Fehler, der bei mir aufgetreten ist entfernen können.
Merge doch mal den aktuellen develop in dein Feature und guck ob es dann klappt.

Reply:
Die Weiterleitung würde ich zur Sicherheit auch drin lassen. Man weiß ja nie, von wo das vielleicht noch aufgerufen wird...

Reply:
Funktioniert jetzt ohne Probleme. Kein Fehler mehr. 👍